mobilecoind/api/proto/mobilecoind_api.proto

```diff
@@ -803,6 +810,40 @@ message GetBlockResponse {
 
     // TxOuts in the block.
     repeated external.TxOut txos = 4;
+
+    // Timestamp result code
+    watcher.TimestampResultCode timestamp_result_code = 5;
+
+    // Timestamp (only valid if timestamp_result_code is TimestampFound)
+    uint64 timestamp = 6;
+}
+
+// Get multiple block datas coupled with timestamps
+message GetBlocksDataRequest {
+    // List of block numbers to query
+    repeated uint64 blocks = 1;
+}
+
+message BlockDataWithTimestamp {
+    /// Whether we found a block for the provided index.
+    bool found = 1;
```

Comment:
:question: I'm not quite following the `found` flag.
It seems like where it's used in the block_provider/src/mobilecoind.rs this gets mapped to `None`
Then in `fog/ledger/server/src/block_service.rs` the None is ignored with a `flatten()` call.

I can understand the desire to say this piece of the request failed, but right now one has to map the response and the request together in order to understand what wasn't found. It may be repeating the block index in the response so one can quickly see that block index 325 wasn't found versus trying to map to a request, not sure :shrug:

Reply:
You're right that right now someone has to map the request to the response, but its easy to add the block index to the response so I'll do that.
I think having the `mobilecoind` API return a response that has a 1:1 mapping with the block indices requested (instead of skipping not found blocks) could make the caller's life easier if they ever wanted to zip the returned data with some other information. That's not required for the block_provider/`fog/ledger/server/src/block_service.rs` stuff so in that case omitting the not found results would be equivalent, but perhaps some other caller in the future might benefit from this. 

Adding the index will make it similar to requests such as https://github.com/mobilecoinfoundation/mobilecoin/blob/main/fog/api/proto/ledger.proto#L155-L161 or https://github.com/mobilecoinfoundation/mobilecoin/blob/main/fog/api/proto/ledger.proto#L253-L278